docs/reference/remap.cue

```diff
@@ -65,4 +64,205 @@ remap: {
 			description: "Raised when the provided input cannot be parsed."
 		}
 	}
+
+	// TRL type system
+	types: [TypeName=string]: {
+		#Use: "parameter" | "return"
+
+		description: string
+		use: [#Use, ...#Use]
+	}
+
+	types: {
+		"any": {
+			description: "A stand-in for any type."
+			use: ["parameter"]
+		}
```

Comment:
This might just be for documentation purposes, but I want to make clear that we don't actually have an `any` type.

The type system resolves the result of an expression to _one or more_ types at compile-time. "any" in our case means it resolves to the entire list of types we support.

To give a quick example:

```rust
.foo
```

This program can resolve to one of:

* string
* boolean
* integer
* float
* map
* array
* null
* timestamp
* regexp

Which can be considered "any", but it's not a concrete type we use internally.

As soon as you apply more constraints to an expression, we narrow the types it can resolve to, e.g.:

```rust
.foo * 5 
```

This can resolve to one of:

* string
* integer
* float

This is also a fallible expression (whereas the previous one is always infallible), _if_ we can't deduce that `.foo` resolves to one of the above types.